### ISSUE TITLE
Does the Manifold plugin work with IntelliJ Community Edition?

### ISSUE THREAD

**Issue by kdurnoga** [manifold-systems/manifold#121]
A bit of a noob question but does the Manifold IntelliJ plugin work with the Community Edition of IntelliJ?
I've got the plugin installed, cloned the example project, and it compiles fine and kinda runs but the IntelliJ editor still complains about various syntax errors (e.g. the preprocessor directives do not get recognized). This happens on IntelliJ 2019.2, Manifold plugin 2019.1.14, and JDK 1.8.0_144.

**Comment by rsmckinney** [comment]
@kdurnoga. Hi.  Yes, the pluign works with IJ Community Edition.  If you are referring to the [manifold-sample-project](https://github.com/manifold-systems/manifold-sample-project), it is built with Java 11, perhaps you need your project's JDK to 11?  You can also change the pom file to use Java 8 if you prefer.  If so, please read the [docs](https://github.com/manifold-systems/manifold/tree/master/manifold-core-parent/manifold#building) concerning pom file setup.

![image](https://user-images.githubusercontent.com/588843/65193634-9c796e80-da30-11e9-9bf9-efd95303a68b.png)

**Comment by kdurnoga** [comment]
I've adjusted the pom file as well as the project settings for JDK 8. The build step under IntelliJ succeeds (so does mvn compile) and I'm able to run the project. (I've also tried switching to JDK 12, which I happen to have installed, and the sample project crashes the compiler with the stacktrace identical to the one given here https://bugs.openjdk.java.net/browse/JDK-8228646 .)
Sadly, code inspection seems to be separated from the building step and oblivious to Manifold plugin presence. It still displays multiple syntax errors.
Is there a way to debug the plugin itself, or at least to see if it is triggered during code inspection?

**Comment by rsmckinney** [comment]
Can you share your project on github or other means?  I can help diagnose the problem, hopefully just a build issue.

**Comment by kdurnoga** [comment]
Sorry for postponing my response for so long but up until now I did not have spare time to investigate this issue. I'm unable to reopen the issue but I'm letting you know, @rsmckinney, what I've found 'cause you might still be interested in fixing it.

tl;dr: The problem occurs deterministically on Windows if project's dependencies (the manifold jar, to be precise) reside on a different drive than the IntelliJ installation directory (to be exact, the working directory of a JVM instance running IJ which defaults to the installation folder, unless specified otherwise manually). For instance, I keep my local Maven repo with `manifold-ij-2019.3.8.jar` on drive `C:` but have a fresh copy of IJ installed to drive `D:`. In this case, the manifold plugin fails to recognize a project that uses the Manifold library as a Manifold-dependent project and simply skips it without displaying errors.

The above behavior is caused by the following handling of the Manifold jar file by the Manifold plugin: 
```
VirtualFile path = ... // this is the path to the manifold jar dependency file
URL url = new URL(path.getUrl());
File file = new File(url.getFile());
File absFile = file.getAbsoluteFile();
```
The pitfall is that `url.getFile()` does not extract the authority part of the url. Put differently, the drive name is dropped from the path, so the path becomes directory-relative, but some drive name is prepended back to the path by calling `file.getAbsoluteFile()`. To do this, the working directory of the JVM is used (cf. https://github.com/openjdk/jdk/blob/master/src/java.base/windows/classes/java/io/WinNTFileSystem.java#L326 ) and thus a path to a non-existing file may be generated by swapping the proper drive name with the drive name of the working directory.

The quickest way to reproduce this problem is to force the working directory of the IJ to be on a different drive than the Manifold jar, e.g. by setting `-Duser.dir=D:/` in `java64.exe.vmoptions` file.

**Comment by rsmckinney** [comment]
Fixed with https://github.com/manifold-systems/manifold-ij/commit/4141bbc7b88f01b9ef3ea4776d3dab1919f1224d (manifold ij plugin codebase)

Fix is available in next plugin release, which is awaiting Jetbrains approval, usually within two business days of submission.

Thanks for reporting this issue!
